psi4/driver/driver.py

```diff
@@ -873,6 +873,150 @@ def properties(*args, **kwargs):
         return core.variable('CURRENT ENERGY')
 
 
+def optimize_geometric(name, **kwargs):
+
+    # will move imports to top of file when i'm sure where this method goes
+    import qcengine as qcng
+    import qcelemental as qcel
+    import geometric
+
+    if '/' in name:
+        raise ValidationError(f'Composite methods like {name} are not yet supported with geomeTRIC')
+                                 
+    # Collect all user-specified keywords
+    keywords = {}
+    for opt in core.get_global_option_list():
+        if core.has_global_option_changed(opt):
+            keywords[opt] = core.get_global_option(opt)
+    basis = keywords['BASIS']
+
+    # Make sure the molecule the user provided is the active one
+    molecule = kwargs.get('molecule', core.get_active_molecule())
+
+    # If we are freezing cartesian, do not orient or COM
+    if core.get_local_option("OPTKING", "FROZEN_CARTESIAN"):
+        molecule.fix_orientation(True)
+        molecule.fix_com(True)
+    molecule.update_geometry()
+
+    input_data = {
+        "keywords": {
+            "convergence_set": core.get_global_option("G_CONVERGENCE"),
+            "maxiter": core.get_global_option("GEOM_MAXITER"),
+            "coordsys": "tric",
+            "enforce": 0.1,
+            "constraints": {},
+            "program": "psi4",
+        },
+        "input_specification": {
+            "driver": "gradient",
+            "model": {
+                "method": name,
+                "basis" : basis,
+            },
+            "keywords" : keywords,
+        },
+        "initial_molecule": molecule.to_schema(dtype=2),
+    }
+
+    if input_data["keywords"]["convergence_set"] in ["CFOUR", "QCHEM", "MOLPRO"]:
+        input_data["keywords"]["convergence_set"] = "GAU_TIGHT"
+
+    # Parse JSON
+    input_opts = geometric.run_json.parse_input_json_dict(input_data)
+    M, engine = geometric.optimize.get_molecule_engine(**input_opts)
```

Comment:
You will want to build your own engine, see [here](https://github.com/leeping/geomeTRIC/blob/master/geometric/engine.py#L846). In addition you can build your own custom Molecule conversion class which is commended I think.

Reply:
Thanks, using a custom engine simplified things a lot